src/ray/raylet/worker_pool.cc

```diff
@@ -53,19 +53,19 @@ namespace ray {
 
 namespace raylet {
 
-/// A constructor that initializes a worker pool with num_workers workers for
-/// each language.
-WorkerPool::WorkerPool(boost::asio::io_service &io_service, int num_workers,
-                       int maximum_startup_concurrency, int min_worker_port,
-                       int max_worker_port, std::shared_ptr<gcs::GcsClient> gcs_client,
-                       const WorkerCommandMap &worker_commands,
-                       const std::unordered_map<std::string, std::string> &raylet_config,
-                       std::function<void()> starting_worker_timeout_callback)
+WorkerPool::WorkerPool(
```

Comment:
Some code blocks are not related to job ID. So it's not easy to simplify all the if-else blocks. But as I said before, the new code is compatible with existing jobs except for some config changes (the ones now in job configs). As long as no two jobs are deployed to the same cluster, the new code should work just fine. So my suggestion actually is to remove the flag.

@ericl You may have concerns about the worker-capping feature. The new code has no design change in resource management. CPU is still a node-level resource, not a per-node-per-job resource. Let's say that the `num_cpus` is set to 8 on a node, and we have two jobs on this node each has already occupied 3 and 4 CPUs, then there's no way to schedule a new actor/task which requires 2 CPUs, for any job.

We are actively testing the multi-tenancy feature with different kinds of jobs on one cluster. And this is not the final PR about multi-tenancy. We definitely shouldn't try to adopt this feature (by running multiple jobs into one cluster) unless it's fully production-ready. Maybe I should rename the `job_config` parameter of `ray.init()` to `_job_config` for now, just like `_internal_config`.

Reply:
@kfstorm that scenario doesn't make sense, workers should have nothing to do with the resource management. If we need workers to launch a task, we should create new ones, and kill idle workers to keep the total excess usage under control.

Reply:
Ideally, we can do that, and we should do that. But for now, as long as we don't run multiple jobs in the same cluster, it should be OK to turn multi-tenancy feature on by default. That's why I suggest that we can actually remove the feature flag since @stephanie-wang is worried about the complexity of if-elses.

Reply:
I'm not really worried about the if-elses; just wanted to see if there was a way to avoid them.